### PR TITLE
Preparation for Flint 3

### PR DESCRIPTION
**Pull request by Joel-Dahne**
Flint 3 is getting closer, an [https://groups.google.com/g/flint-devel/c/xc5lFBU9a94](alpha-version) was tagged about a month ago. With this release the Arb package will now be a part of Flint instead of being a separate package. That means that we have to make some minor updates to the package.

The main changes in this PR are:
- Use `libflint` from `FLINT_jll` instead of `libarb` from `Arb_jll`
- Update the poly structs according to the change in Flint. This is a breaking change in the sense that it changes the order of the memory layout, but is unlikely to matter when coding in Julia.
- Update the parser and reparse the documentation, based on the `flint-3.0.0-alpha1` tag of Flint. This adds a few new methods but also removes quite a lot of methods which are now based on generic versions in Flint.

There is still no `FLINT_jll` for Flint 3 so the tests won't be able to run here. To run it locally you need to build an appropriate version of Flint and then add the line
```
[FLINT_jll]
libflint_path = "/path/to/flint2/libflint.so"
```
to the file `LocalPreferences.toml` at the root of this repository. 

Once Flint 3 is release and `FLINT_jll` is updated we can move forward with this PR.

### CODE REVIEW

**Comment by kalmarek** [comment]
great for taking care of this!

the changes seem to be pretty minimal, given the changes in arb/flint. That's a big plus!

**Comment by Joel-Dahne** [comment]
Now `FLINT_jll` has been updated to version 3, which means this PR is ready. However, I want to do some more testing to see if everything works well before merging it.

**Comment by Joel-Dahne** [comment]
I did some more testing and everything seems to work well! This is then ready for merging after #172

**Comment by kalmarek** [comment]
@Joel-Dahne You are on a spree :)

**Comment by Joel-Dahne** [comment]
You are doing a good job in keeping up!  

This should be ready to be merged now. Then I have one more thing I want to add, `union` and `intersection` for polynomials and `midpoint` for `Acb`. After that I think we could be ready for a 1.0 release?

**Review by kalmarek — COMMENTED** [review]
I had a look brief look at this (again).
It seems that almost all changes are `libarb` → `libflint` rename?
Or is there something else hidden?

File: src/Arblib.jl [hunk 1]
```diff
@@ -27,12 +27,13 @@ export Mag,
...
+macro libflint(function_name)
+    return (:($function_name), libflint)
 end
 
-macro libflint(function_name)
-    return (:($function_name), Arb_jll.libflint)
+# For backwards compatibility with when Arb was a separate package
+macro libarb(function_name)
+    return (:($function_name), libflint)
 end
```
[the remark this answers is not in the document]
Comment (Joel-Dahne) [reply src/Arblib.jl]: Good point! I removed it now, since we will probably not do any other release before  1.0

**Comment by Joel-Dahne** [comment]
> I had a look brief look at this (again). It seems that almost all changes are `libarb` → `libflint` rename? Or is there something else hidden?

That is basically it. The only other main change is the reordering for the poly structs, but that you don't really have to care about from Julia. Then it is just documentation changes and making use of some of the newly added functions in Flint.

**Review by kalmarek — APPROVED** [review]

**Pull request merged**